q2_assembly/bowtie2/tests/test_utils.py

```diff
@@ -27,6 +27,11 @@ def test_get_subdir_from_path_contig(self):
         exp = "sample1"
         self.assertEqual(obs, exp)
 
+    def test_get_subdir_from_path_contig_underscores(self):
+        obs = _get_subdir_from_path("/path/to/dir/sample_1_contigs.fa")
+        exp = "sample_1"
+        self.assertEqual(obs, exp)
```

Comment:
I suggest a few more tests of this. The last one is pretty pathological, but we might as well support it. This requires the update to the logic that I suggested above.

```suggestion
        self.assertEqual(_get_subdir_from_path("/path/to/dir/sample_1_contigs.fa"), "sample_1")
        self.assertEqual(_get_subdir_from_path("/path/to/dir/sample1_contigs.fa"), "sample1")
        self.assertEqual(_get_subdir_from_path("/path/to/dir/s_am-p_le_1_contigs.fa"), "s_am-p_le_1")
        self.assertEqual(_get_subdir_from_path("/path/to/dir/_contigs.fa_contigs.fa"), "_contigs.fa")
```